src/flag_gems/ops/embedding.py

```diff
@@ -70,11 +70,15 @@ def embedding_backward_kernel(
     if not HAS_PADDING_IDX:
         grad_in += row_idx * N
         embedding_grad = tl.load(grad_out + cols, mask, other=0.0)
+        if tl.constexpr(embedding_grad.dtype.is_bf16()):
+            embedding_grad = embedding_grad.to(tl.float32)
         tl.atomic_add(grad_in + cols, embedding_grad, mask=mask)
```

Comment:
Atomically updating gradients from a massive number of ctas may be a problem. Could we reduce contention by incorporating local update with a gsl style kernel?